source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+    } else if (mShowScrollGuide) {
+#else
+    // Don't transition to the next state until the animation is finished.
+    // Otherwise, the "View Map" text would change to "Back to Mario" too early.
+    } else if (mShowScrollGuide && !mLayout.isAnime(ANIM_OUT_UNDER)) {
+#endif
+        mStateMgrScrollGuide.changeState(StateID_ScrollGuideOnStageAnimeEndCheck);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ScrollGuideOnStageWait() {
+    mShowScrollGuide = false;
+}
+
+void dCourseSelectGuide_c::initializeState_ScrollGuideOnStageAnimeEndCheck() {
+    mExtension = Remocon::EXTENSION_UNKNOWN;
+    mInMapView = true;
+    mpNullPanes[N_guideViewC_00]->setVisible(false);
+    mpNullPanes[N_guideViewR_01]->setVisible(false);
+    mpNullPanes[N_left_00]->setVisible(false);
+    mLayout.AnimeStartSetup(ANIM_IN_UNDER, false);
+    mScrollGuideRelated = true;
+}
+void dCourseSelectGuide_c::executeState_ScrollGuideOnStageAnimeEndCheck() {
+    if (mEndScrollGuide) {
+        mLayout.AnimeEndSetup(ANIM_IN_UNDER);
+        mStateMgrScrollGuide.changeState(StateID_ScrollGuideExitAnimeEndCheck);
+    } else if (!mLayout.isAnime(-1)) {
+        mStateMgrScrollGuide.changeState(StateID_ScrollGuideDisp);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ScrollGuideOnStageAnimeEndCheck() {}
+
+void dCourseSelectGuide_c::initializeState_ScrollGuideDisp() {
+    mLayout.LoopAnimeStartSetup(ANIM_LOOP_UNDER);
+}
+void dCourseSelectGuide_c::executeState_ScrollGuideDisp() {
+    if (mEndScrollGuide) {
+        mStateMgrScrollGuide.changeState(StateID_ScrollGuideExitAnimeEndCheck);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ScrollGuideDisp() {}
+
+void dCourseSelectGuide_c::initializeState_ScrollGuideExitAnimeEndCheck() {
+    mEndScrollGuide = false;
+    mLayout.AnimeEndSetup(ANIM_LOOP_UNDER);
+    mLayout.AnimeStartSetup(ANIM_OUT_UNDER, false);
+}
+void dCourseSelectGuide_c::executeState_ScrollGuideExitAnimeEndCheck() {
+    if (!mLayout.isAnime(-1)) {
+        mStateMgrScrollGuide.changeState(StateID_ScrollGuideOnStageWait);
+    }
+}
+void dCourseSelectGuide_c::finalizeState_ScrollGuideExitAnimeEndCheck() {
+    mExtension = Remocon::EXTENSION_UNKNOWN;
+    mScrollGuideRelated = false;
+    mpNullPanes[N_guideViewC_00]->setVisible(true);
+    mpNullPanes[N_guideViewR_01]->setVisible(true);
+    mpNullPanes[N_left_00]->setVisible(true);
+    mInMapView = false;
+}
+
+void dCourseSelectGuide_c::UpdateGuide(short courseNo, dWmLib::CourseType_e type) {
+    if (mCourseNo == courseNo) {
+        return;
+    }
+    mCourseType = dWmLib::GetCourseTypeFromCourseNo(courseNo);
+    mCourseNo = courseNo;
+    CourseSelectSet(type);
+    if (type != 1) {
+        CollectionCoinSet();
+    }
+    dInfo_c *info = dInfo_c::m_instance;
+    dCyuukan_c *checkpoint;
+    u8 wNo, cNo;
+    cNo = courseNo;
+    wNo = mWorldNo;
+    mpPicturePanes[P_flagSkull_00]->setVisible(false);
+    checkpoint = &info->mCyuukan;
+    for (int i = 0; i < 2; i++) {
```

Comment:
Possible to use an enum value for this? Kinda pedantic i know...

Reply:
Alternatively, a macro `CHECKPOINT_COUNT`?